app/src/main/java/org/wikipedia/random/RandomItemFragment.java

```diff
@@ -139,8 +142,11 @@ public void onGlobalLayout() {
                 extractView.getViewTreeObserver().removeOnGlobalLayoutListener(this);
             }
         });
+
         imageView.loadImage(TextUtils.isEmpty(summary.getThumbnailUrl()) ? null
-                : Uri.parse(summary.getThumbnailUrl()));
+                : Uri.parse((WikipediaApp.getInstance().isTablet()
+                ? ImageUrlUtil.getUrlForPreferredSize(summary.getThumbnailUrl(), PREFERRED_TABLET_THUMBNAIL_SIZE)
```

Comment:
Any reason we can't just use 800dp in all cases?  The image looks kind of pixelated on phone displays, too...

Reply:
Yes, we can!